### PR TITLE
fix(cosh-ng): [types] drop cross-workspace dev-dep

### LINKED ISSUE #1973: [nightly][cosh-ng] cosh-ng RPM BUILD_FAILED: cosh-types dev-dependency path escapes workspace root (since b3e6f8f9)
## 问题

cosh-ng RPM 构建在 nightly-20260729-020133 worker 上 BUILD_FAILED。

`bash scripts/rpm-build.sh cosh-ng` 在 cosh-types 加载 manifest 阶段失败:

```
error: failed to load manifest for workspace member `cosh-ng-0.13.0/crates/cosh-types`
referenced by workspace at `cosh-ng-0.13.0/Cargo.toml`

Caused by:
  failed to load manifest for dependency `ws-ckpt-common`

Caused by:
  failed to read `/root/anolisa/scripts/rpmbuild/BUILD/ws-ckpt/src/crates/common/Cargo.toml`

Caused by:
  No such file or directory (os error 2)
```

## 根因

commit b3e6f8f9 (2026-07-28) `fix(cosh-ng): [types,platform] fix wire errors` 给 `crates/cosh-types/Cargo.toml` 加了 dev-dependency:

```toml
[dev-dependencies]
ws-ckpt-common = { path = "../../../ws-ckpt/src/crates/common" }
```

该 path 从 `cosh-ng/crates/cosh-types/` 出发,3 个 `..` 跨出 cosh-ng workspace 根目录,指向 monorepo 内的 `anolisa/src/ws-ckpt/src/crates/common/`(在 monorepo 内有效,本地 `cargo test --locked -p cosh-types` 能跑通)。

但 `scripts/rpm-build.sh` 的 `build_cosh_ng` 用 `tar -cf - -C "$COSH_DIR" .` 打包只 include cosh-ng 子树,不 include ws-ckpt/。rpmbuild 解包到 `BUILD/cosh-ng-0.13.0/`,path 解析后指向 `BUILD/ws-ckpt/src/crates/common/` —— 该目录在 tarball 外不存在。

cargo 在 `cargo build --workspace --release` 时计算 Cargo.lock 需要为所有 workspace 成员(含 cosh-types)的 dev-dependency 加载 manifest(即使 dev-dep 不被编译进 release binary,只用于 `tests/checkpoint_wire_conformance.rs`),因此 path 失败 → build failed。

## 影响

- nightly-20260729-020133 cosh-ng worker BUILD_FAILED,case_total=0
- 后续每晚 nightly 都会撞这个 bug,直到修复
- cosh-ng 0.13.0 RPM 无法构建,blocked cosh-ng-tests 100 用例套件接入 nightly 的整个 plan

## 复现步骤(产品侧手工操作)

```bash
# 1. clone anolisa monorepo
git clone https://github.com/alibaba/anolisa
cd anolisa

# 2. 单独构建 cosh-ng RPM
bash scripts/rpm-build.sh cosh-ng

# 看到:error: failed to read /root/anolisa/scripts/rpmbuild/BUILD/ws-ckpt/src/crates/common/Cargo.toml
```

## 修复建议

两层修法,治本程度递增:

### 短期(rpm-build.sh vendor)
PR #1971 在 `scripts/rpm-build.sh` `build_cosh_ng` 创建 tarball 时额外 vendor `ws-ckpt-common` 源码作为 tarball 顶层 sibling dir `ws-ckpt/`,并 sed 过滤原始 ws-ckpt workspace Cargo.toml 的 members 行只保留 `crates/common`(避免拉 daemon/cli 进一步的 deps)。`%setup -q` 解包后 BUILD/ 下同时有 `cosh-ng-0.13.0/` 和 `ws-ckpt/`,path 正确解析。

### 长期(治本,二选一)
1. 把 wire protocol types 提取成独立 crate 发到 crates.io(anolisa monorepo 多 cargo workspace 通过 path dep 互相引用是结构性问题,RPM 构建只打包单 workspace 子树时必撞)
2. 把 `tests/checkpoint_wire_conformance.rs` 移到 `ws-ckpt` 仓内(因为该 test 验证 cosh-types 与 ws-ckpt-common wire format 一致,ws-ckpt 仓更适合做这个 conformance test)

### 失败的尝试(供参考)
- `[dev-dependencies] ws-ckpt-common = { path=..., optional=true }` + `[features] wire-conformance = ["ws-ckpt-common"]` —— cargo 1.91 报错 `dev-dependencies are not allowed to be optional`
- `[dependencies] ws-ckpt-common = { path=..., optional=true }` + feature gate —— cargo lockfile 生成时仍加载 optional dep manifest 记录版本,path 解析依旧失败

## 验证

ECS 47.239.57.210 上跑 PR #1971 后:
```
[OK] cosh-ng RPM built successfully
  cosh-ng-0.13.0-2.alnx4.x86_64.rpm  (7.9M)
  cosh-ng-0.13.0-2.alnx4.src.rpm  (1.9M)
```

附:同时发现 cosh-core 通过 reqwest → hyper-tls → native-tls → openssl-sys 拉 openssl-sys,缺 `openssl-devel` 会 build failed(spec.in 虽 `BuildRequires: openssl-devel` 但 rpm-build.sh 用 `rpmbuild --nodeps` 不自动装)—— 这个在 agentic-os-tests 仓 cosh-ng-tests SKILL.md Step 1 单独修(显式 `yum install -y openssl-devel`)。

## 关联
- Fix PR: #1971
- 引入 commit: b3e6f8f9 `fix(cosh-ng): [types,platform] fix wire errors`
- nightly run: nightly-20260729-020133

### PR DESCRIPTION
**Pull request by samchu-zsl**
## Description

Replace the cross-workspace `ws-ckpt-common` path dev-dependency in `cosh-types` with a checked-in golden fixture of bincode wire bytes. The path dependency (introduced by #1514) escapes the cosh-ng workspace root, so RPM builds that pack only the cosh-ng subtree failed during cargo lockfile resolution, breaking every nightly cosh-ng RPM build. The fixture was captured while the previous conformance suite still proved cosh-types byte-identical to the authoritative `ws-ckpt-common`, so the byte-level lock is anchored to a verified-consistent state. Compile-time exhaustiveness is preserved via per-variant case-name matches; regeneration is gated behind `UPDATE_WIRE_GOLDENS=1` for intentional, daemon-coordinated protocol changes.

Known trade-off (accepted by maintainer): drift on the ws-ckpt side is no longer caught at cosh-ng build time; runtime checkpoint integration coverage remains the backstop. Once merged, the tarball vendoring workaround in #1971 can be reverted.

## Related Issue

closes #1973

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional change)
- [ ] Performance improvement
- [ ] CI/CD or build changes

## Scope

- [ ] `cosh` (copilot-shell)
- [x] `cosh-ng` (cosh-ng)
- [ ] `sec-core` (agent-sec-core)
- [ ] `skill` (os-skills)
- [ ] `sight` (agentsight)
- [ ] `tokenless` (tokenless)
- [ ] `ckpt` (ws-ckpt)
- [ ] `memory` (agent-memory)
- [ ] `anolisa` (anolisa-cli)
- [ ] `skillfs` (SkillFS)
- [ ] Multiple / Project-wide

## Checklist

- [x] I have read the Contributing Guide
- [x] My code follows the project's code style
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the documentation accordingly
- [x] For `cosh-ng`: `cargo clippy --all-targets -- -D warnings` and `cargo fmt --check` pass
- [x] Lock files are up to date (`package-lock.json` / `Cargo.lock`)

## Testing

```bash
cd src/cosh-ng
cargo fmt --all -- --check                                        # clean
cargo clippy --workspace --all-targets -- -D warnings             # clean
cargo test --package cosh-types                                   # 27 passed
cargo test --package cosh-platform checkpoint                     # 22 passed
cargo test --package cosh-cli --test cli_integration checkpoint   # 19 passed
cargo test --workspace -- --test-threads=4                        # all green except 2 pre-existing
                                                                  # cosh-shell raw_cli PTY flakes
                                                                  # (pass serially, unrelated crate)
cargo doc --workspace --no-deps                                   # clean
```

RPM failure-mode regression: copied the cosh-ng subtree to an isolated directory (simulating the rpmbuild tarball) — `cargo metadata --locked` now resolves successfully; on `main` it fails with `failed to read .../ws-ckpt/src/crates/common/Cargo.toml`.

## Additional Notes

`Cargo.lock` loses exactly the 18 lines added by #1514 for the dev-dependency. The `test-necessity-registry.tsv` entry `types-wire-conformance` still matches the rewritten test (same path pattern, same contract).

### CODE REVIEW

**Comment by samchu-zsl** [comment]
> [!CAUTION]
> **This PR was generated and submitted by AI.**
> Please review all changes carefully before merging. Pay special attention to logic correctness, edge cases, and potential side effects.

File: src/cosh-ng/crates/cosh-types/tests/checkpoint_wire_conformance.rs [hunk 1]
```diff
@@ -1,268 +1,132 @@
+//! Byte-level golden lock for the ws-ckpt IPC wire protocol.
```
[the remark this answers is not in the document]
Comment (samchu-zsl) [reply src/cosh-ng/crates/cosh-types/tests/checkpoint_wire_conformance.rs:1]: Fixed in revision 381f2435: `fixture_path()` now derives the file name from `PROTOCOL_VERSION` (`checkpoint-wire-v{PROTOCOL_VERSION}.json`), making the constant the single source of truth. A version bump without renaming the fixture file now fails loudly with the missing-fixture panic that names the expected path.

File: src/cosh-ng/crates/cosh-types/tests/checkpoint_wire_conformance.rs [hunk 2]
```diff
@@ -815,15 +343,107 @@ fn local_response_samples() -> Vec<local::WsCkptResponse> {
...
 #[test]
-fn every_response_variant_matches_authoritative_bincode() {
-    for value in local_response_samples() {
-        assert_same_bincode(&value, &wire_response(&value));
+fn checkpoint_wire_matches_golden_fixture() {
+    let requests = request_cases();
+    let errors = error_cases();
+    let responses = response_cases();
+
+    let mut expected = encode_cases(&requests);
```
[the remark this answers is not in the document]
Comment (samchu-zsl) [reply src/cosh-ng/crates/cosh-types/tests/checkpoint_wire_conformance.rs:396]: Not a real issue: the duplicate-name `BTreeSet` check guards the generated sample list, and the fixture side is covered by the strict ordered equality assertion `assert_eq!(fixture_names, expected_names)` a few lines below — any hand-edited duplicate, insertion, deletion, or reordering in `checkpoint-wire-v1.json` makes the name vectors unequal and fails that assertion. A separate duplicate scan over fixture cases would be redundant with the full-sequence comparison.

**Review by KaiLongZhou — COMMENTED** [review]
LGTM 

用 checked-in golden fixture（bincode 字节 hex） 替代跨 workspace dev-dep，是短期 vendor（#1971）和长期发 crates.io 之间的最优解：
彻底消除跨 workspace path → RPM 失败模式根除（PR 作者做了隔离目录 cargo metadata --locked 回归验证）
Cargo.lock 精确回退 #1514 引入的 18 行；顺带掉的 js-sys/wasm-bindgen 是 chrono feature-union 收缩的正常副产物，无害

**Review by KaiLongZhou — APPROVED** [review]
